### ISSUE TITLE
Profanity `0.4.6` segfaults on connecting to Slack.com

### ISSUE THREAD

**Issue by cipriancraciun** [boothj5/profanity#599]
It seems that the latest Profanity (`0.4.6` from the ArchLinux repositories), seems to segfault when connecting to a Slack.com "domain".  (For the record it works fine with GMail.)

Any ideas what might be the problem?

The "redacted" (replacing the confidential information with `whatever`) log is:

    03/08/2015 18:56:22: xmpp: DBG: sock_connect to whatever.xmpp.slack.com:5222 returned 4
    03/08/2015 18:56:22: xmpp: DBG: attempting to connect to whatever.xmpp.slack.com
    03/08/2015 18:56:22: prof: INF: Starting main event loop
    03/08/2015 18:56:22: xmpp: DBG: connection successful
    03/08/2015 18:56:22: conn: DBG: SENT: <?xml version="1.0"?><stream:stream to="whatever.xmpp.slack.com" xml:lang="en" version="1.0" xmlns="jabber:client" xmlns:stream="http://etherx.jabber.org/streams">
    03/08/2015 18:56:22: xmpp: DBG: RECV: <stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' id='1974871016228916943-0' from='whatever.xmpp.slack.com' version='1.0'>
    03/08/2015 18:56:22: xmpp: DBG: RECV: <stream:features>   <starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls">    <required/>  </starttls></stream:features>
    03/08/2015 18:56:22: conn: DBG: SENT: <starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>
    03/08/2015 18:56:22: xmpp: DBG: RECV: <proceed xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>
    03/08/2015 18:56:22: xmpp: DBG: handle proceedtls called for proceed
    03/08/2015 18:56:22: xmpp: DBG: proceeding with TLS
    03/08/2015 18:56:23: conn: DBG: SENT: <?xml version="1.0"?><stream:stream to="whatever.xmpp.slack.com" xml:lang="en" version="1.0" xmlns="jabber:client" xmlns:stream="http://etherx.jabber.org/streams">
    03/08/2015 18:56:23: xmpp: DBG: RECV: <stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' id='1974871016228916943-1' from='whatever.xmpp.slack.com' version='1.0'>
    03/08/2015 18:56:23: xmpp: DBG: RECV: <stream:features>  <mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl">    <mechanism>PLAIN</mechanism>  </mechanisms></stream:features>

**Comment by boothj5** [comment]
Thanks for the info, I recently saw the same problem with Slack.

It seems the Slack XMPP gateway has been updated and some of the XMPP stanza's contain whitespace between elements, which causes the segfault in libstrophe (https://github.com/strophe/libstrophe).

I've added a fix to libstrophe.  Until Profanity 0.4.7 is released, if you want to use Slack, you'll need to build and install libstrophe from the latest code to get the fix.

If you have the time to try this let me know your findings, otherwise I'll close this issue, since I've tested against Slack using my work account.
